File: backend/app/models/tracking.py
from sqlalchemy import (
    Column,
    Float,
    ForeignKey,
    Integer,
    String,
)

from app.core.database import Base


class TrackedEntity(Base):

    __tablename__ = (
        "tracked_entities"
    )

    id = Column(
        Integer,
        primary_key=True,
        index=True,
    )

    job_id = Column(
    Integer,
        ForeignKey(
            "video_jobs.id"
        ),
    )

    tracking_id = Column(
        Integer,
        index=True,
    )

    first_seen = Column(
        String
    )

    confidence = Column(
        Float
    )

    status = Column(
        String
    )


class TrackingFrame(Base):

    __tablename__ = (
        "tracking_frames"
    )

    id = Column(
        Integer,
        primary_key=True,
        index=True,
    )

    job_id = Column(
        Integer,
        ForeignKey(
            "video_jobs.id"
        ),
    )

    timestamp = Column(
        Float,
        index=True,
    )

    tracking_id = Column(
        Integer
    )

    x = Column(Float)

    y = Column(Float)

    width = Column(Float)

    height = Column(Float)

    confidence = Column(
        Float
    )

File: frontend/src/types/tracking.ts
export type IdentityStatus =

  | "active"
  | "occluded"
  | "exited";


export interface PersistentIdentity {

  id: number;

  tracking_id: number;

  first_seen: string;

  confidence: number;

  status: IdentityStatus;

  last_seen?: string;

  job_id?: string;
}


export interface TrackingBox {

  id: number;

  x: number;

  y: number;

  width: number;

  height: number;

  confidence: number;
}


export interface TrackingFrame {

  timestamp: number;

  boxes: TrackingBox[];
}